container/lxd/network.go

```diff
@@ -0,0 +1,181 @@
+		if s.networkAPISupport {
+			return errors.Annotate(s.ensureDefaultBridge(profile), "ensuring default bridge config")
+		}
+		return errors.Errorf("unexpected LXD %q profile without eth0 device: %+v", profile.Name, profile)
+	}
+
+	netName := eth0["parent"]
+
+	if s.networkAPISupport {
+		net, _, err := s.GetNetwork(netName)
+		if err != nil {
+			return errors.Annotatef(err, "retrieving network %q", netName)
+		}
+		if err := ensureNoIPv6(net); err != nil {
+			return errors.Trace(err)
+		}
+	}
+
+	if eth0["type"] != "nic" || eth0["nictype"] != "bridged" {
+		return errors.Errorf("eth0 is not configured as a bridge in %q profile: %v", profile.Name, eth0)
+	}
+
+	s.localBridgeName = netName
+	logger.Infof("LXD %q profile uses network bridge %q", profile.Name, netName)
+
+	// If this LXD version supports the network API, we are done.
+	// Otherwise we need to check the legacy lxd-bridge config file.
+	if s.networkAPISupport {
+		return nil
+	}
+	return errors.Trace(checkBridgeConfigFile(ioutil.ReadFile))
+}
+
+// ensureDefaultBridge ensures that the default LXD bridge exists,
+// that it is not configured to use IPv6, and that the eth0 device exists in
+// the input profile.
+// An error is returned if the bridge exists with IPv6 configuration.
+// If the bridge does not exist, it is created.
+func (s *Client) ensureDefaultBridge(profile *api.Profile) error {
+	net, eTag, err := s.GetNetwork(network.DefaultLXDBridge)
+	if err != nil {
+		if !isLXDNotFound(err) {
+			return errors.Trace(err)
+		}
+		req := api.NetworksPost{
+			Name: network.DefaultLXDBridge,
+			NetworkPut: api.NetworkPut{Config: map[string]string{
+				"ipv6.address": "none",
+				"ipv6.nat":     "false",
+			}},
+		}
+		err := s.CreateNetwork(req)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		net, eTag, err = s.GetNetwork(network.DefaultLXDBridge)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	} else {
+		if err := ensureNoIPv6(net); err != nil {
+			return errors.Trace(err)
+		}
+	}
+
+	s.localBridgeName = network.DefaultLXDBridge
+
+	// Add the eth0 device with the bridge as its parent.
+	nicType := "macvlan"
+	if net.Type == "bridge" {
+		nicType = "bridged"
+	}
+	profile.Devices["eth0"] = map[string]string{
+		"nictype": nicType,
+		"parent":  network.DefaultLXDBridge,
+	}
+	return errors.Trace(s.UpdateProfile(profile.Name, profile.Writable(), eTag))
+}
+
+// ensureNoIPv6 checks that the input network has no IPv6 configuration.
+// An error is returned when it does.
+func ensureNoIPv6(net *api.Network) error {
+	if !net.Managed {
+		return nil
+	}
+	cfg, ok := net.Config["ipv6.address"]
+	if !ok {
+		return nil
+	}
+	if cfg == "none" {
+		return nil
+	}
+
+	return errors.Errorf("juju does not support IPv6. Disable IPv6 in LXD via:\n"+
+		"\tlxc network set %s ipv6.address none\n"+
+		"and run the command again", net.Name)
+}
+
+const BridgeConfigFile = "/etc/default/lxd-bridge"
+
+func checkBridgeConfigFile(reader func(string) ([]byte, error)) error {
```

Comment:
This was just moved code, right? Did you change this function at all? (Assuming it was just moved, I didn't read it)

Reply:
Yes. No change to the body there.